client/components/recycleAndReuseComponents/Steps/StepThree/Geolocation/utils.js

```diff
@@ -97,44 +116,58 @@ export const getNearestNonBlueBinFacilities = (items, { lat, lng }) => {
 	const results = [];
 
 	for (const item of items) {
-		const validLocations = [];
+		let distance = null;
+		let nearestLocation = null;
 
 		for (const place of physicalChannels) {
 			if (
 				place.categories_accepted.includes(item.category) &&
 				place.type.includes(item.condition)
 			) {
-				validLocations.push({
-					id: place.id,
-					postal: place.postcode,
-					distance:
-						Math.round(
-							calcCrow(
-								lat,
-								lng,
-								place.latitude,
-								place.longitude,
-							) * 100,
-						) / 100,
-					latitude: place.latitude,
-					longitude: place.longitude,
-					address: place.address,
-					channel_name: place.channel_name,
-					operating_hours: place.operating_hours,
-					contact: place.contact,
-					website: place.website,
-					categories_accepted: place.categories_accepted,
-					organisation_name: place.organisation_name,
-					type: place.type,
-				});
+				let calcDist = Math.round(
+					calcCrow(
+						lat,
+						lng,
+						place.latitude,
+						place.longitude,
+					) * 100,
+				) / 100;
+				if (!distance || (distance && distance > calcDist)) {
+					distance = calcDist;
+					nearestLocation = setNearestLocation(place);
+					nearestLocation.distance = calcDist;
+				}
 			}
 		}
+		const itemNames = [];
+		itemNames.push(item.description);
+		if (nearestLocation) {
+			nearestLocation.items = itemNames;
+			results.push(nearestLocation);
+		} else {
+			//For handling items which do not have a valid location
+		}
+	}
 
-		sortByDistance(validLocations);
-		const nearestLocation = validLocations[0];
-		nearestLocation.items = item.description;
-		results.push(nearestLocation);
+	const final = [];
+
+	for (const result of results) {
+		if (final.length == 0) {
+			final.push(result);
+		} else {
+			let found = false;
+			for (const item of final) {
```

Comment:
can rename this item? because quite a few items floating around, and with `results` and `final`, it's almost at the limit of ambiguous variables.

don't worry too much about the name being too verbose.

Reply:
Alright, will rename to avoid ambiguity.